molo/surveys/tests/test_views.py

```diff
@@ -374,6 +374,31 @@ def test_can_only_see_sites_surveys_in_admin(self):
             '<h2><a href="/admin/surveys/submissions/%s/">'
             'Test Survey</a></h2>' % molo_survey_page.pk)
 
+    def test_changing_languages_changes_survey(self):
+        # Create a survey
+        self.user = self.login()
+        molo_survey_page, molo_survey_form_field = \
+            self.create_molo_survey_page_with_field(parent=self.surveys_index)
+        # Create a translated survey
+        response = self.client.post(reverse(
+            'add_translation', args=[molo_survey_page.id, 'fr']))
+        translated_survey = MoloSurveyPage.objects.get(
+            slug='french-translation-of-test-survey')
+        translated_survey.save_revision().publish()
+        create_molo_survey_formfield(
+            survey=translated_survey, field_type='singleline',
+            label="Your favourite animal in french", required=True)
+
+        # when requesting the english survey with the french language code
+        # it should return the french survey
```

Comment:
Maybe add a test that this also works when the survey has been submitted. I found that the issue mainly occurs after the user submits and answer to the survey

Reply:
@sewagodimo the redirect does not check if the user has submitted or not, it redirects them based on the language they are on versus the language they have requested